php-checks/src/main/java/org/sonar/php/checks/ReturnOfBooleanExpressionCheck.java

```diff
@@ -41,69 +44,50 @@
 
 @SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.READABILITY)
 @SqaleConstantRemediation("2min")
-public class ReturnOfBooleanExpressionCheck extends SquidCheck<LexerlessGrammar> {
-  @Override
-  public void init() {
-    subscribeTo(
-      PHPGrammar.IF_STATEMENT,
-      PHPGrammar.ALTERNATIVE_IF_STATEMENT);
-  }
+public class ReturnOfBooleanExpressionCheck extends PHPVisitorCheck {
 
-  @Override
-  public void visitNode(AstNode astNode) {
-    if (!hasElseIf(astNode) && hasElse(astNode) && returnsBoolean(getTrueStatement(astNode)) && returnsBoolean(getFalseStatement(astNode))) {
-      getContext().createLineViolation(this, "Replace this \"if-then-else\" statement by a single \"return\" statement.", astNode);
-    }
-  }
+  public static final String KEY = "S1126";
+  private static final String MESSAGE = "Replace this \"if-then-else\" statement by a single \"return\" statement.";
 
-  private static AstNode getTrueStatement(AstNode ifStmt) {
-    return getInnerStmtOfIfStmt(ifStmt.getFirstChild(PHPGrammar.ELSE_CLAUSE, PHPGrammar.ALTERNATIVE_ELSE_CLAUSE)
-      .getFirstChild(PHPGrammar.STATEMENT, PHPGrammar.INNER_STATEMENT_LIST));
-  }
+  @Override
+  public void visitIfStatement(IfStatementTree tree) {
+    super.visitIfStatement(tree);
 
-  private static AstNode getFalseStatement(AstNode ifStmt) {
-    return getInnerStmtOfIfStmt(ifStmt.getFirstChild(PHPGrammar.STATEMENT, PHPGrammar.INNER_STATEMENT_LIST));
-  }
+    boolean elseClauseReturnsBoolean = tree.elseClause() != null && returnsBoolean(getStatement(tree.elseClause().statements()));
+    boolean mainClauseReturnsBoolean = returnsBoolean(getStatement(tree.statements()));
 
-  private static AstNode getInnerStmtOfIfStmt(AstNode statement) {
-    if (statement != null && statement.is(PHPGrammar.INNER_STATEMENT_LIST)) {
-      return statement.getNumberOfChildren() > 1 ? null : statement.getFirstChild();
+    if (tree.elseifClauses().isEmpty() && mainClauseReturnsBoolean && elseClauseReturnsBoolean) {
```

Comment:
@vilchik-elena Minor performance improvement: we currently always evaluate all 3 boolean expressions, but we could avoid that.